modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/validator/AMImageValidatorImpl.java

```diff
@@ -75,19 +81,112 @@ public boolean isValid(FileVersion fileVersion) {
 			return false;
 		}
 
+		if (!_isFileVersionStoredMetadataSupported(fileVersion)) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private boolean _checkDDMFormFieldsMapValue(
+		List<DDMFormFieldValue> ddmFormFieldValues,
+		long imageToolImageMaxValue) {
+
+		DDMFormFieldValue ddmFormFieldValue = ddmFormFieldValues.get(0);
+
+		Value value = ddmFormFieldValue.getValue();
+
+		Long length = Long.valueOf(value.getString(value.getDefaultLocale()));
+
+		if (length <= 0) {
```

Comment:
Can we simplify the condition to
```
if ((length <= 0) || (length >= imageToolImageMaxValue)) {
    return false;
}
else {
    return true;
}
```

Reply:
In the original code conditions return different boolean values. so I don't think that we could create a unique clause.

Anyway, checking this code with fresh eyes it looks weird and it seems that I did a mistake here. I think that the first clause should be:

`imageToolImageMaxValue <= 0` (the configured value) instead of `length <= 0`(the image metadata values)

I'll recheck it.